engineyard/ey-cookbooks-dev-v7
# [FBZ-10498] clamAV Support

Description of your patch
-------------
Adds ClamAV anti-malware support as an optional package for a stack-v7 environment.

Recommended Release Notes
-------------
Adds ClamAV support for EY Stack-v7. Upon this release, clamAV could be configured via environment-variables for given instance roles of an environment.

Estimated risk
-------------
Medium

Components involved
-------------
`ey-clamav` , `ey-base`

Dependencies
-------------
`ey-base`

Description of testing done
-------------
1. Booted an environment (Solo with util)
2. Uploaded the patch as an overlay.
3. Setup `EY_CLAMAV_ENABLED` to `true` and added paths `EY_CLAMAV_SOLO_PATHS` to `["/data", "/tmp"]` , verified if the cronjobs appeared in the solo instance including the freshclam KB update job which should run weekly.
4. Added path for `EY_CLAMAV_UTIL_PATHS` as an array, again applied chef for the whole environment and checked if cronjobs are available on `util` instance.

QA Instructions
-------------

* On Solo instance
1. Upload the patch to the environment
2. Boot the solo environment with a new sidekiq instance
3. Enable clamAV and setup paths for scanner cron as per the Readme
4. Check if crons appear in the 2 specific instances.

* On multi-environment
1. Upload the patch to the environment
2. Boot the multi environment with 2 application instances, db master and slave and a util instance (of any name)
3. Enable clamAV and setup paths for scanner cron as per the Readme
4. Check if crons appear in the specified instances.
5. Remove a few paths and check if the changes are reflected on the respective instances.
